code/game/objects/effects/spiders.dm

```diff
@@ -156,7 +156,7 @@
 	pixel_y = rand(6,-6)
 	START_PROCESSING(SSobj, src)
 	//50% chance to grow up
-	if(prob(50))
+	if(!istype(src, /obj/effect/spider/spiderling/non_growing) && prob(50))

```

Comment:
Generally speaking if you are having to check the type of `src` then you should be doing something else. In this case it looks like you want `amount_grown != -1` instead, as `non_growing` and various other 'do not grow up' procs like the vermin event use `-1` to indicate non-growing spiderlings.